File: backend/src/main/java/com/rabbit/app/modules/cage/dto/CageSummary.java
package com.rabbit.app.modules.cage.dto;

import java.math.BigDecimal;
import java.util.Date;
import java.util.List;

public class CageSummary {
    private Long cageId;
    private String cageNumber;
    private Integer rabbitCount;
    private Boolean isFed;

    private Date lastFeedTime;
    private String lastFeedType;
    private BigDecimal lastFeedAmount;
    private String lastFeedUnit;

    private Integer abnormalUndealCount;
    private Date lastAbnormalTime;
    private String lastAbnormalStatus;

    private List<CageRabbitBrief> rabbits;

    public Long getCageId() {
        return cageId;
    }

    public void setCageId(Long cageId) {
        this.cageId = cageId;
    }

    public String getCageNumber() {
        return cageNumber;
    }

    public void setCageNumber(String cageNumber) {
        this.cageNumber = cageNumber;
    }

    public Integer getRabbitCount() {
        return rabbitCount;
    }

    public void setRabbitCount(Integer rabbitCount) {
        this.rabbitCount = rabbitCount;
    }

    public Boolean getIsFed() {
        return isFed;
    }

    public void setIsFed(Boolean fed) {
        isFed = fed;
    }

    public Date getLastFeedTime() {
        return lastFeedTime;
    }

    public void setLastFeedTime(Date lastFeedTime) {
        this.lastFeedTime = lastFeedTime;
    }

    public String getLastFeedType() {
        return lastFeedType;
    }

    public void setLastFeedType(String lastFeedType) {
        this.lastFeedType = lastFeedType;
    }

    public BigDecimal getLastFeedAmount() {
        return lastFeedAmount;
    }

    public void setLastFeedAmount(BigDecimal lastFeedAmount) {
        this.lastFeedAmount = lastFeedAmount;
    }

    public String getLastFeedUnit() {
        return lastFeedUnit;
    }

    public void setLastFeedUnit(String lastFeedUnit) {
        this.lastFeedUnit = lastFeedUnit;
    }

    public Integer getAbnormalUndealCount() {
        return abnormalUndealCount;
    }

    public void setAbnormalUndealCount(Integer abnormalUndealCount) {
        this.abnormalUndealCount = abnormalUndealCount;
    }

    public Date getLastAbnormalTime() {
        return lastAbnormalTime;
    }

    public void setLastAbnormalTime(Date lastAbnormalTime) {
        this.lastAbnormalTime = lastAbnormalTime;
    }

    public String getLastAbnormalStatus() {
        return lastAbnormalStatus;
    }

    public void setLastAbnormalStatus(String lastAbnormalStatus) {
        this.lastAbnormalStatus = lastAbnormalStatus;
    }

    public List<CageRabbitBrief> getRabbits() {
        return rabbits;
    }

    public void setRabbits(List<CageRabbitBrief> rabbits) {
        this.rabbits = rabbits;
    }
}

File: flutter_app/lib/src/domain/models/cage_summary.dart
class CageSummary {
  const CageSummary({
    required this.cageId,
    required this.cageNumber,
    required this.rabbitCount,
    required this.isFed,
    required this.lastFeedTime,
    required this.lastFeedType,
    required this.lastFeedAmount,
    required this.lastFeedUnit,
    required this.abnormalUndealCount,
    required this.lastAbnormalTime,
    required this.lastAbnormalStatus,
  });

  final int cageId;
  final String cageNumber;
  final int rabbitCount;
  final bool isFed;
  final DateTime? lastFeedTime;
  final String lastFeedType;
  final double? lastFeedAmount;
  final String lastFeedUnit;
  final int abnormalUndealCount;
  final DateTime? lastAbnormalTime;
  final String lastAbnormalStatus;

  static CageSummary fromJson(Map<String, dynamic> json) {
    return CageSummary(
      cageId: _intValue(json['cageId']),
      cageNumber: json['cageNumber'] as String? ?? '',
      rabbitCount: _intValue(json['rabbitCount']),
      isFed: _boolValue(json['isFed']),
      lastFeedTime: _dateValue(json['lastFeedTime']),
      lastFeedType: json['lastFeedType'] as String? ?? '',
      lastFeedAmount: _doubleValue(json['lastFeedAmount']),
      lastFeedUnit: json['lastFeedUnit'] as String? ?? '',
      abnormalUndealCount: _intValue(json['abnormalUndealCount']),
      lastAbnormalTime: _dateValue(json['lastAbnormalTime']),
      lastAbnormalStatus: json['lastAbnormalStatus'] as String? ?? '',
    );
  }

  static int _intValue(Object? value) {
    if (value is int) return value;
    if (value is num) return value.toInt();
    if (value is String) return int.tryParse(value) ?? 0;
    return 0;
  }

  static double? _doubleValue(Object? value) {
    if (value is num) return value.toDouble();
    if (value is String) return double.tryParse(value);
    return null;
  }

  static bool _boolValue(Object? value) {
    if (value is bool) return value;
    if (value is num) return value != 0;
    return value?.toString().toLowerCase() == 'true';
  }

  static DateTime? _dateValue(Object? value) {
    if (value is int) return DateTime.fromMillisecondsSinceEpoch(value);
    if (value is num) {
      return DateTime.fromMillisecondsSinceEpoch(value.toInt());
    }
    if (value is String) return DateTime.tryParse(value);
    return null;
  }
}
